### PR TITLE
PointNet 1.0 implementation.

### PR DESCRIPTION
**Pull request by ataiya**

### CODE REVIEW

**Comment by ataiya** [comment]
Also added Charles Qi to the contributors so he can review the pull request.

**Comment by ataiya** [comment]
@julienvalentin before I merge, I let you take a look at the copybara check?

File: tensorflow_graphics/nn/layer/pointnet.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    self.channels = channels
+    self.momentum = momentum
+
+  def build(self, input_shape):
+    """Builds the layer with a specified input_shape."""
+    self.conv = Conv2D(self.channels, (1, 1), input_shape=input_shape)
+    self.bn = BatchNormalization(momentum=self.momentum)
+
+  # pylint: disable=arguments-differ
+  def call(self, inputs, training):
```
Comment (jackd) [inline tensorflow_graphics/nn/layer/pointnet.py:76]: Request: can `training` default to `None` like the rest of keras? `BatchNormalization` knows how to handle it :)

File: tensorflow_graphics/nn/layer/pointnet.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+      channels: the number of generated feature.
+      momentum: the momentum of the batch normalization layer.
+    """
+    super(PointNetConv2Layer, self).__init__()
+    self.channels = channels
+    self.momentum = momentum
+
+  def build(self, input_shape):
+    """Builds the layer with a specified input_shape."""
+    self.conv = Conv2D(self.channels, (1, 1), input_shape=input_shape)
```
Comment (jackd) [inline tensorflow_graphics/nn/layer/pointnet.py:72]: Is there a reason for implementing this as a conv layer and not just `Dense`?

Comment (ataiya) [reply tensorflow_graphics/nn/layer/pointnet.py:72]: I had an alternative branch just using Dense, but then the structure of the batch normalization got more complicated. I needed a bunch of reshapes before/after the batch norm. And if you dig deeper, this is because (sadly) batch norm receives in input a feature axis and *not* a batch axis. The second reason is that Charles' original PointNet implementation was following this design, and this is supposed to be as close as possible to it as it can be.

File: tensorflow_graphics/nn/layer/pointnet.py (hunk 1 above)
Comment (ataiya) [reply tensorflow_graphics/nn/layer/pointnet.py:76]: yes, I meant to fix this, let me take a look, but you have a good point

Comment (ataiya) [reply tensorflow_graphics/nn/layer/pointnet.py]: Ok, now I recall. It's funny, because a regular keras.Layer has:
```def call(self, inputs, **kwargs):```
while a BatchNormalization
```def call(self, inputs, training=None):```

I poked a bit deeper, and the second pattern seems a good match

Comment (ataiya) [reply tensorflow_graphics/nn/layer/pointnet.py]: yet this is still given linter warnings without the disable...
```
def call(self, inputs, training=None):  # pylint: disable=arguments-differ
```
